internal/nginx/runtime/manager.go

```diff
@@ -65,8 +67,22 @@ func (m *ManagerImpl) Reload(ctx context.Context) error {
 	return nil
 }
 
-func findMainProcess(readFile readFileFunc) (int, error) {
-	content, err := readFile(pidFile)
+func findMainProcess(checkFile checkFileFunc, readFile readFileFunc, timeout time.Duration) (int, error) {
+	startTime := time.Now()
+	deadline := startTime.Add(timeout)
+
+	fileCheck := func() (content []byte, err error) {
+		for time.Now().Before(deadline) {
+			_, err := checkFile(pidFile)
+			if err == nil {
+				return readFile(pidFile)
+			}
+			time.Sleep(1 * time.Second)
```

Comment:
better to only continue with the error return is file not exist. This way we will fail faster in case of some  other errors, like related to permissions.